commit d2132f453e3308adc82ab7c101bd5220a9a34167
Author: Mel Gorman <mgorman@techsingularity.net>
Date:   Thu Jan 23 22:11:13 2025 +0000

    mm: security: Allow default HARDENED_USERCOPY to be set at compile time
    
    HARDENED_USERCOPY defaults to on if enabled at compile time. Allow
    hardened_usercopy= default to be set at compile time similar to
    init_on_alloc= and init_on_free=. The intent is that hardening
    options that can be disabled at runtime can set their default at
    build time.
    
    Signed-off-by: Mel Gorman <mgorman@techsingularity.net>
    Link: https://lore.kernel.org/r/20250123221115.19722-3-mgorman@techsingularity.net
    Signed-off-by: Kees Cook <kees@kernel.org>

diff --git a/mm/usercopy.c b/mm/usercopy.c
index 16d63bd010af..e65a612ce8e8 100644
--- a/mm/usercopy.c
+++ b/mm/usercopy.c
@@ -255,7 +255,8 @@ void __check_object_size(const void *ptr, unsigned long n, bool to_user)
 }
 EXPORT_SYMBOL(__check_object_size);
 
-static bool enable_checks __initdata = true;
+static bool enable_checks __initdata =
+		IS_ENABLED(CONFIG_HARDENED_USERCOPY_DEFAULT_ON);
 
 static int __init parse_hardened_usercopy(char *str)
 {